packages/code-du-travail-frontend/src/conventions/texte/SummaryTitle.js

```diff
@@ -1,37 +1,48 @@
 import React from "react";
 import PropTypes from "prop-types";
 import SummaryItem from "./SummaryItem";
-import styled from "styled-components";
+import { TocListItem, TocList, TocLink } from "./toc";
 
-const SummaryTitle = ({ type, data, children, expanded, onToggleExpanded }) => {
+const SummaryTitle = ({
+  type,
+  data,
+  children,
+  expanded,
+  onToggleExpanded,
+  tocbotEnabled
+}) => {
   const { id, titre } = data;
 
+  if (!children || children.length == 0) {
+    return (
+      <TocListItem type={type} id={id} level={0}>
+        <TitleLink type={type} id={id}>
+          {titre}
+        </TitleLink>
+      </TocListItem>
+    );
+  }
   return (
-    <Wrapper title={`${type} ${id}`}>
-      <div>
-        <Title>
-          {children && children.length > 0 ? (
-            <TitleLink
-              href="#"
-              onClick={e => {
-                e.preventDefault();
-                onToggleExpanded(id, !expanded);
-              }}
-            >
-              {titre}&nbsp;{expanded ? "▲" : "▼"}
-            </TitleLink>
-          ) : (
-            <TitleLink href={`#${id}`}>{titre}</TitleLink>
-          )}
-        </Title>
-      </div>
-      {expanded &&
-        children.map((child, idx) => (
-          <SummaryItem key={idx} level={0} {...child}>
-            {child.children}
-          </SummaryItem>
-        ))}
-    </Wrapper>
+    <TocListItem type={type} id={id} level={0}>
+      <TitleLink
+        id={id}
+        type={type}
+        onClick={e => {
+          e.preventDefault();
```

Comment:
to avoid passing a new function to TitleLink to each render, 
you can use a useCallback hook here
```
 const clickHandler = useCallback(event => {
  event.preventDefault();
  onToggleExpanded(id, !expanded);
}, [data.id, expanded])
```

Reply:
thanks I didn't know about useCallback, I refactored using it https://github.com/SocialGouv/code-du-travail-numerique/commit/98fe141df2102405c3924d457582f02b75d3941f